src/py/flwr/common/record/configsrecord.py

```diff
@@ -58,27 +58,57 @@ def is_valid(__v: ConfigsScalar) -> None:
 
 
 class ConfigsRecord(TypedDict[str, ConfigsRecordValues]):
-    """Configs record."""
+    """Configs record.
+
+    A :code:`ConfigsRecord` is a Python dictionary designed to ensure that
+    each key-value pair adheres to specified data types.
+
+    Parameters
+    ----------
+    configs_dict : Optional[Dict[str, ConfigsRecordValues]]
+        A dictionary that stores basic types (i.e. `str`, `int`, `float`, `bytes` as
+        defined in `ConfigsScalar`) and lists of such types (see
+        `ConfigsScalarList`).
+    keep_input : bool (default: True)
+        A boolean indicating whether config passed should be deleted from the input
+        dictionary immediately after adding them to the record. When set
+        to True, the data is duplicated in memory. If memory is a concern, set
+        it to False.
+
+    Examples
+    --------
+    The usage of a :code:`ConfigsRecord` is envisioned for storing Python built-in
+    types such as :code:`int`, :code:`float`, :code:`str`, :code:`bytes` and lists of
+    these. All types allowed are defined in
+    :code:`common.typing.ConfigsRecordValues`. While lists are supported, we
```

Comment:
Should this be
```suggestion
    :code:`flwr.common.typing.ConfigsRecordValues`. While lists are supported, we
```

Reply:
exporting `ConfigsRecordValues` and `MetricsRecordValues` in https://github.com/adap/flower/pull/4052

Reply:
#4052 merged